src/frequenz/client/dispatch/test/generator.py

```diff
@@ -0,0 +1,138 @@
+                self._rng.randint(0, 59) for _ in range(self._rng.randint(0, 10))
+            ],
+            byhours=[self._rng.randint(0, 23) for _ in range(self._rng.randint(0, 10))],
+            byweekdays=[
+                self._rng.choice(list(Weekday)[1:])
+                for _ in range(self._rng.randint(0, 7))
+            ],
+            bymonthdays=[
+                self._rng.randint(1, 31) for _ in range(self._rng.randint(0, 10))
+            ],
+            bymonths=[
+                self._rng.randint(1, 12) for _ in range(self._rng.randint(0, 12))
+            ],
+        )
+
+    def generate_dispatch(self) -> Dispatch:
+        """Generate a random dispatch instance.
+
+        Returns:
+            a random dispatch instance
+        """
+        self._last_id += 1
+        create_time = datetime.fromtimestamp(
+            self._rng.randint(0, 1000000), tz=timezone.utc
+        )
+
+        return Dispatch(
+            id=self._last_id,
+            create_time=create_time,
+            update_time=create_time + timedelta(seconds=self._rng.randint(0, 1000000)),
+            microgrid_id=self._rng.randint(0, 100),
+            type=str(self._rng.randint(0, 100_000)),
+            start_time=datetime.now().astimezone(timezone.utc)
+            + timedelta(seconds=self._rng.randint(0, 1000000)),
+            duration=timedelta(seconds=self._rng.randint(0, 1000000)),
+            selector=self._rng.choice(  # type: ignore
+                [
+                    self._rng.choice(list(ComponentCategory)[1:]),
+                    [
+                        self._rng.randint(1, 100)
+                        for _ in range(self._rng.randint(1, 10))
+                    ],
+                ]
+            ),
+            active=self._rng.choice([True, False]),
+            dry_run=self._rng.choice([True, False]),
+            payload={
+                f"key_{i}": self._rng.choice(
+                    [
+                        self._rng.randint(0, 100),
+                        self._rng.uniform(0, 100),
+                        self._rng.choice([True, False]),
+                        self._rng.choice(["a", "b", "c"]),
+                    ]
+                )
+                for i in range(self._rng.randint(0, 10))
+            },
+            recurrence=self.generate_recurrence_rule(),
+        )
+
+
+def to_create_params(dispatch: Dispatch) -> dict[str, Any]:
+    """Convert a dispatch to client.create parameters.
+
+    Args:
+        dispatch: The dispatch to convert.
+
+    Returns:
+        dict[str, Any]: The create parameters.
+    """
+    return {
+        "microgrid_id": dispatch.microgrid_id,
+        "_type": dispatch.type,
+        "start_time": dispatch.start_time,
+        "duration": dispatch.duration,
+        "selector": dispatch.selector,
+        "active": dispatch.active,
+        "dry_run": dispatch.dry_run,
+        "payload": dispatch.payload,
+        "recurrence": dispatch.recurrence,
+    }
```

Comment:
Minor, but I would move this to the `_client.py` module (considering you move the `FakeClient` there, as this is more releated to the client than to dispatches generation.